src/rust/rust-proto/src/protocol/status.rs

```diff
@@ -129,7 +129,7 @@ impl From<GrpcCode> for Code {
 #[derive(thiserror::Error)]
 pub struct Status {
     /// The gRPC status code, found in the `grpc-status` header.
-    code: Code,
+    pub code: Code,
```

Comment:
Maybe a getter `pub fn code() -> &Code`? I've started doing that for new code in rust-proto (in the branch that I haven't PRed yet). Making this `pub` is fine for now, though.

Reply:
could we match on that?

Reply:
Yes, I believe so?